pages/api/indexer/stats/count_club_domains.ts

```diff
@@ -18,51 +16,129 @@ export default async function handler(
-    case "99": // according to the enlightened despot, this club is called "99" even though it contains 100 domains
-      regex = /^\d{2}\.stark$/;
-      break;
-    case "999":
-      regex = /^\d{3}\.stark$/;
-      break;
-    case "10k":
-      regex = /^\d{4}\.stark$/;
-      break;
+  const dbOutput = (
+    await domainCollection
+      .aggregate([
+        {
+          $match: {
+            "_chain.valid_to": null,
+            creation_date: {
+              $gte: new Date(beginTime),
+            },
+          },
+        },
+        {
+          '$group': {
+            '_id': {
+              '$cond': [
+                {
+                  '$regexMatch': {
+                    'input': '$domain',
+                    'regex': /^.\.stark$/
+                  }
+                }, 'single_letter', {
+                  '$cond': [
+                    {
+                      '$regexMatch': {
+                        'input': '$domain',
+                        'regex': /^\d{2}\.stark$/
+                      }
+                    }, '99', {
+                      '$cond': [
+                        {
+                          '$regexMatch': {
+                            'input': '$domain',
+                            'regex': /^.{2}\.stark$/
+                          }
+                        }, 'two_letters', {
+                          '$cond': [
+                            {
+                              '$regexMatch': {
+                                'input': '$domain',
+                                'regex': /^\d{3}\.stark$/
+                              }
+                            }, '999', {
+                              '$cond': [
+                                {
+                                  '$regexMatch': {
+                                    'input': '$domain',
+                                    'regex': /^.{3}\.stark$/
+                                  }
+                                }, 'three_letters', {
+                                  '$cond': [
+                                    {
+                                      '$regexMatch': {
+                                        'input': '$domain',
+                                        'regex': /^\d{4}\.stark$/
+                                      }
+                                    }, '10k', 'none'
+                                  ]
+                                }
+                              ]
+                            }
+                          ]
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ]
+            },
+            'count': {
+              '$sum': 1
+            }
+          },
+        },
+        {
+          $project: {
+            _id: 0,
+            club: "$_id",
+            count: "$count",
+          },
+        },
+      ])
+      .toArray()
+  );
+
+
+  let _99;
+  let _999;
+  for (const doc of dbOutput) {
```

Comment:
Je comprend pas comment fonctionne ce doc.club, tu peux m'expliquer ?

Reply:
L'idée c'est que je demande à la bdd de ne s'intéresser qu'aux domaines qui ont été mintés après la date "since", ensuite elle va parcourir les résultats et leur construire un identificateur. Cet identificateur c'est le nom du club associé à la regex qui va match et none sinon. Elle va grouper les résultats par identificateur et compter la taille de chaque groupe (c'est le .count). Finalement je projette ma db pour renommer l'identificateur en un field club et c'est quasiment fini. Il ne reste plus qu'à additionner les groupes complémentaires (genre le 99 club doit être ajouté au two_letters club).